tests/integration/metrics_test.go

```diff
@@ -228,3 +234,167 @@ func TestExitedContainersWithoutTTL(t *testing.T) {
 		assert.Empty(t, sample)
 	})
 }
+
+const (
+	relativePathToTestdataFileystem = "testdata/cgroupsV1_host/"
+	InspectorContainerID            = "my-container"
+	InspectorPID                    = 666
+)
+
+var mockedTimeForAllMetricsTest = time.Date(2022, time.January, 1, 4, 3, 2, 0, time.UTC)
+
+func TestAllMetricsPresent(t *testing.T) {
+	expectedSample := biz.Sample{
+		Pids: biz.Pids{
+			Current: InspectorPID,
+			Limit:   8000,
+		},
+		Network: biz.Network{
+			RxBytes:   3402,
+			RxDropped: 2,
+			RxErrors:  10,
+			RxPackets: 28,
+			TxBytes:   5,
+			TxDropped: 2,
+			TxErrors:  3,
+			TxPackets: 4,
+		},
+		BlkIO: biz.BlkIO{
+			TotalReadCount:  39,
+			TotalWriteCount: 89,
+			TotalReadBytes:  2387968,
+			TotalWriteBytes: 50,
+		},
+		CPU: biz.CPU{
+			CPUPercent:       0.15003804623431827,
+			KernelPercent:    1.19999999,
+			UserPercent:      100,
+			UsedCores:        2.8546433726,
+			LimitCores:       float64(runtime.NumCPU()),
+			UsedCoresPercent: float64(100) * 2.8546433726 / float64(runtime.NumCPU()),
+			ThrottlePeriods:  2384,
+			ThrottledTimeMS:  96.578349164,
+			Shares:           1024,
+		},
+		Memory: biz.Memory{
+			UsageBytes:            11620352,
+			CacheUsageBytes:       2310144,
+			RSSUsageBytes:         11620352,
+			MemLimitBytes:         104857600,
+			UsagePercent:          11.08203125,
+			KernelUsageBytes:      724992,
+			SwapUsageBytes:        1620352,
+			SwapOnlyUsageBytes:    18446744073699551616,
+			SwapLimitBytes:        209715200,
+			SwapLimitUsagePercent: 0.77264404296875,
+			SoftLimitBytes:        262144000,
+		},
+		RestartCount: 2,
+	}
+
+	// Create a tempDir that will be the root of our mocked filsystem
+	hostRoot := t.TempDir()
+
+	// Create our mocked container cgroups filesystem in the tempDir directory
+	// that will be a symLink to our cgroups testdata
+	cgroupsFolder := filepath.Join(hostRoot, "cgroup")
+	mockFilesystem, err := filepath.Abs(filepath.Join(relativePathToTestdataFileystem, "cgroup"))
+	require.NoError(t, err)
+
+	err = os.Symlink(mockFilesystem, cgroupsFolder)
+	require.NoError(t, err)
+
+	// Create mocked directory proc
+	err = os.Mkdir(filepath.Join(hostRoot, "proc"), 0755)
+	require.NoError(t, err)
+
+	err = generateMockedProcMountsFile(cgroupsFolder, hostRoot)
+	require.NoError(t, err)
+
+	err = generateMockedProcPIDCGroupFile(hostRoot)
+	require.NoError(t, err)
+
+	err = genereateMockedProcNetDevFile(hostRoot)
+	require.NoError(t, err)
```

Comment:
nit: perhaps it would clean up the test code if this gets moved to another func WDYT?
It is a bit hard to understand what is part of the mock creation and what is the test itself.

Reply:
done